dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java

```diff
@@ -246,15 +246,15 @@ default SourceFile getSourceFileByPath(long entryId, String tag, DescriptorLangu
         checkOptionalAuthRead(user, entry);
 
         // tighten permissions for hosted tools and workflows
-        if (!entry.getIsPublished()) {
-            if (entry instanceof Tool && ((Tool)entry).getMode() == ToolMode.HOSTED) {
-                throw new CustomWebApplicationException("Entry not published", HttpStatus.SC_FORBIDDEN);
-            }
-            if (entry instanceof Workflow && ((Workflow)entry).getMode() == WorkflowMode.HOSTED) {
-                throw new CustomWebApplicationException("Entry not published", HttpStatus.SC_FORBIDDEN);
-            }
-        }
         if (!user.isPresent() || AuthenticatedResourceInterface.userCannotRead(user.get(), entry)) {
+            if (!entry.getIsPublished()) {
+                if (entry instanceof Tool && ((Tool)entry).getMode() == ToolMode.HOSTED) {
```

Comment:
Ideally the previous 2 if statements will catch everything, e.g., if we someday support private GitHub repos. Just a comment, no action required.